File: src/main/java/com/wesker926/leetcode/algorithms/p0240/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0240.s1;

/**
 * @author wesker.gh
 * @date 2021/10/25
 * @description BST
 */
public class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int l = matrix[0].length - 1, r = 0;
        while (l >= 0 && r < matrix.length) {
            if (matrix[r][l] == target) {
                return true;
            } else if (matrix[r][l] > target) {
                l--;
            } else {
                r++;
            }
        }
        return false;
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0240/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/10/25
// @description: BST
package s1

func searchMatrix(matrix [][]int, target int) bool {
	l, r := len(matrix[0])-1, 0
	for l >= 0 && r < len(matrix) {
		if matrix[r][l] == target {
			return true
		} else if matrix[r][l] > target {
			l--
		} else {
			r++
		}
	}
	return false
}
